app/models/concerns/with_qualifications.rb

```diff
@@ -69,5 +63,10 @@ def qualifications
     def qualifications_description
       qualifications.map(&:upcase).sort.join(" with ")
     end
+
+    def qualification=(value)
+      super(value)
+      self.profpost_flag = qts? ? :recommendation_for_qts : :postgraduate
```

Comment:
`qts?` that bit looks really magically guess it comes from 

```ruby
enum qualification: %i[qts pgce_with_qts pgde_with_qts pgce pgde]
```

Reply:
yup that's right